src/routes/home/components/DisciplineCard.tsx

```diff
@@ -49,11 +48,19 @@ interface PropsType {
 }
 
 const DisciplineCard = (props: PropsType): ReactElement => {
-  const { discipline, showProgress, onPress, navigateToNextExercise } = props
-  const { data: progress } = useReadProgress(discipline)
+  const { disciplineId, onPress, navigateToNextExercise } = props
+  const { data: discipline } = useLoadDiscipline(disciplineId)
+  const [progress, setProgress] = useState<number | null>(null)
   const moduleAlreadyStarted = progress !== null && progress !== 0
   const [documents, setDocuments] = useState<Document[] | null>(null)
-  const { data: nextExercise } = useReadNextExercise(discipline)
+  const [nextExercise, setNextExercise] = useState<NextExercise | null>(null)
+
+  useEffect(() => {
+    if (discipline) {
+      AsyncStorage.getProgress(discipline).then(setProgress).catch(reportError)
+      getNextExercise(discipline).then(setNextExercise).catch(reportError)
+    }
+  }, [discipline])
```

Comment:
Why do you use a custom useEffect now instead of useReadProgress/useReadNextExercise? Because discipline may be null?
Not sure if that would be better but we could adjust those hooks to allow null as discipline and just return null if that is the case.

Reply:
You are right, changed it, is much nicer now.